### PR TITLE
WELD-2330 Improve logging in DeferredEventNotification.

### PR DESCRIPTION
**Pull request by manovotn**

### CODE REVIEW

**Comment by manovotn** [comment]
If this looks alright, I'll make a PR for 2.4 as well.

File: modules/jta/src/main/java/org/jboss/weld/module/jta/DeferredEventNotification.java [hunk 1]
```diff
@@ -82,7 +82,13 @@ protected void execute() {
             }.run();
 
         } catch (Exception e) {
-            EventLogger.LOG.asyncObserverFailure(metadata);
+            String stackTrace;
+            if (e.getCause().getMessage() != null) {
```
Comment (mkouba) [inline modules/jta/src/main/java/org/jboss/weld/module/jta/DeferredEventNotification.java]: `e.getCause()` might return null.

File: modules/jta/src/main/java/org/jboss/weld/module/jta/DeferredEventNotification.java [hunk 2]
```diff
@@ -82,7 +82,13 @@ protected void execute() {
...
+                stackTrace = "StackTrace: " + e.getCause().getMessage();
+            } else {
+                stackTrace = "No stack trace to show, the exception's message was empty.";
```
Comment (mkouba) [inline modules/jta/src/main/java/org/jboss/weld/module/jta/DeferredEventNotification.java]: It's actually not a stack trace but a exception/detail message.

**Comment by mkouba** [comment]
Merged, thanks!

**Pull request closed without merging**